ports/openldap/openssl.patch

```diff
@@ -1,13 +1,35 @@
 diff --git a/configure.ac b/configure.ac
-index 626d024..c1eb141 100644
+index 626d024..e97f548 100644
 --- a/configure.ac
 +++ b/configure.ac
-@@ -1235,7 +1235,7 @@ if test $ol_with_tls = openssl || test $ol_with_tls = auto ; then
+@@ -1223,6 +1223,9 @@ fi
  
- 		AC_CHECK_LIB(ssl, SSL_export_keying_material_early,
- 			[have_openssl=yes], [have_openssl=no],
--			[-lcrypto])
-+			[-lcrypto -lpthread])
+ ol_link_tls=no
+ if test $ol_with_tls = openssl || test $ol_with_tls = auto ; then
++	PKG_CHECK_MODULES(OPENSSL, openssl >= 1.1.1, [have_openssl=yes], [AC_MSG_FAILURE([OpenSSL is required])])
++	CFLAGS="${OPENSSL_CFLAGS} ${CFLAGS}"
++	ac_cv_lib_ssl_SSL_export_keying_material_early=yes
+ 	AC_CHECK_HEADERS(openssl/ssl.h)
  
- 		if test $have_openssl = yes ; then
- 			ol_with_tls=openssl
+ 	if test $ac_cv_header_openssl_ssl_h = yes ; then
+@@ -1245,7 +1248,8 @@ if test $ol_with_tls = openssl || test $ol_with_tls = auto ; then
+ 			AC_DEFINE(HAVE_OPENSSL, 1, 
+ 				[define if you have OpenSSL])
+ 
+-			TLS_LIBS="-lssl -lcrypto"
++			TLS_LIBS="${OPENSSL_LIBS}"
++			AC_SUBST(TLS_MODULE, openssl, [TLS pkg-config module])
```

Comment:
Ditto - this is overkill and potentially negates some of the existing checks?

Reply:
You may call it nice-to-have, but OTOH it gives a more expressive picture downstream, when using many dependencies.